app/Http/Controllers/AccountController.php

```diff
@@ -72,4 +74,72 @@ public function updatePage()
 
         return ['html' => $user->userPage->bodyHTML];
     }
+
+    /**
+     * Gets the new order of user profile page elements from request,
+     * and saves it into the database.
+     */
+    public function updateProfileExtrasOrder(HttpRequest $request)
+    {
+        $this->validate($request, [
+            'order' => 'required|array',
+        ]);
+
+        if (!Auth::check()) {
+            abort(403, trans('errors.codes.http-403'));
+        }
+
+        $user = Auth::user();
+        $order = Request::input('order');
+
+        // jQuery's AJAX methods convert ints to strings unfortunately,
+        // so here we convert them back
+        $order = array_map(function ($x) {
+            if (gettype($x) === 'string') {
+                return intval($x);
+            } else {
+                return $x;
+            }
+        }, $order);
+
+        $count = array_fill(1, User::EXTRAS_COUNT, 0);
+
+        foreach ($order as $i) {
+            // Checking whether there are any values in the order array
+            // that are not IDs of profile page blocks.
+            if ($i > User::EXTRAS_COUNT || $i < 1) {
+                return response()
+                    ->json([
+                        'status' => 'error',
+                        'errors' => [
+                            trans('errors.account.profile-order.invalid-id', ['count' => User::EXTRAS_COUNT]),
+                        ],
+                    ]);
+            }
+        }
+
+        // Checking whether the values in the order array are unique,
+        // so we don't get repeating blocks on the profile page.
+        $occurences = array_count_values($order);
```

Comment:
did I miss a check if all pages are included? Not sure if allowing people to remove a section is a good idea.